geode-core/src/main/java/org/apache/geode/internal/util/DriverJarUtil.java

```diff
@@ -35,17 +39,62 @@ public void registerDriver(String driverClassName)
     registerDriverWithDriverManager(d);
   }
 
+  public void deregisterDriver(String driverClassName) throws SQLException {
+    Enumeration<Driver> driverEnumeration = getDrivers();
+    Driver driver;
+    while (driverEnumeration.hasMoreElements()) {
+      driver = driverEnumeration.nextElement();
+      if (compareDriverClassName(driver, driverClassName)) {
+        deregisterDriverWithDriverManager(driver);
+      }
+    }
+  }
+
+  public List<Driver> getRegisteredDrivers() {
+    Enumeration<Driver> drivers = DriverManager.getDrivers();
+    return Collections.list(drivers);
+  }
+
+  public List<String> getRegisteredDriverNames() {
+    List<String> listOfDriverNames = new ArrayList<>();
+    for (Driver driver : getRegisteredDrivers()) {
+      if (driver instanceof DriverWrapper) {
+        listOfDriverNames.add(((DriverWrapper) driver).getWrappedDriverName());
+      } else {
+        listOfDriverNames.add(driver.getClass().getName());
+      }
+    }
+    return listOfDriverNames;
+
+  }
+
   // The methods below are included to facilitate testing and to make the helper methods in this
   // class cleaner
   Driver getDriverInstanceByClassName(String driverClassName)
       throws ClassNotFoundException, IllegalAccessException, InstantiationException {
     return (Driver) ClassPathLoader.getLatest().forName(driverClassName).newInstance();
   }
 
+  Enumeration<Driver> getDrivers() {
+    return DriverManager.getDrivers();
+  }
+
+  boolean compareDriverClassName(Driver driver, String driverClassName) {
+    if (driver instanceof DriverWrapper) {
+      return ((DriverWrapper) driver).getWrappedDriverName().equals(driverClassName);
+    } else {
+      return driver.getClass().getName().equals(driverClassName);
+    }
+  }
+
   void registerDriverWithDriverManager(Driver driver) throws SQLException {
     DriverManager.registerDriver(driver);
   }
 
+  void deregisterDriverWithDriverManager(Driver driver) throws SQLException {
+    DriverManager.deregisterDriver(driver);
+  }
+
   // DriverManager only uses a driver loaded by system ClassLoader
   class DriverWrapper implements Driver {
```

Comment:
Why does this class exist? Could we get rid of it and just have instances of Driver?
If we get rid of it then we do not need the getWrappedDriverName.
If we keep it then each method that implements a Driver method should have the Override method tag added to it.

Reply:
The issue is with the class loader used by the driver manager when register driver is called. Since it wont have any drivers that are dynamically added to the cluster via the deploy gfsh command, the actually instance of Driver would be rejected, so we create this wrapper which is available in the system class loader to wrap around it. Since we used a different class loader to create an instance of the object contained within the wrapper before passing it to the driver manager we then have access to that instance later when jdbc uses our url to retrieve the driver instance from the class loader, even though the driver manager can't technically create an instance of the driver we're trying to use.

We should add override method tags, I didn't realize we had omitted them.